#!/usr/bin/env python
PACKAGE = "plane_slam"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()


ne_method_enum = gen.enum([	gen.const("COVARIANCE_MATRIX", int_t, 0, "COVARIANCE_MATRIX"),
                                gen.const("AVERAGE_3D_GRADIENT", int_t, 1, "AVERAGE_3D_GRADIENT"),
                                gen.const("AVERAGE_DEPTH_CHANGE", int_t, 2, "AVERAGE_DEPTH_CHANGE"),
                                gen.const("SIMPLE_3D_GRADIENT", int_t, 3, "SIMPLE_3D_GRADIENT") ],
                        "An enum to set normal estimation method")

## Organized multiple plane segmentation
gen.add("ne_method", int_t, 0, "Normal e method", 0, edit_method=ne_method_enum)
gen.add("ne_max_depth_change_factor", double_t, 0, "", 0.02, 0.001, 0.5)
gen.add("ne_normal_smoothing_size", double_t, 0, "", 20.0, 1.0, 50.0)
gen.add("min_inliers", int_t, 0, "", 600, 100, 200000)
gen.add("angular_threshold", double_t, 0, "", 3.0, 1.0, 10.0)
gen.add("distance_threshold", double_t, 0, "", 0.02, 0.005, 0.1)
gen.add("project_bounding_points", bool_t, 0, "", False)

exit(gen.generate(PACKAGE, "plane_slam", "OrganizedSegment"))
